tests/test_pg_drop_counter.py

```diff
@@ -0,0 +1,98 @@
+
+    def set_counter(self, dvs, obj_id, attr, val):
+
+        db = swsscommon.DBConnector(swsscommon.ASIC_DB, dvs.redis_sock, 0)
+        ntf = swsscommon.NotificationProducer(db, "SAI_VS_UNITTEST_CHANNEL")
+
+        r = redis.Redis(unix_socket_path=dvs.redis_sock, db=swsscommon.ASIC_DB)
+        rid = r.hget("VIDTORID", obj_id)
+
+        assert rid is not None
+        fvp = swsscommon.FieldValuePairs([(attr, val)])
+        key = rid
+
+        ntf.send("set_stats", key, fvp)
+
+    def populate_asic(self, dvs, val):
+        for obj_id in self.pgs:
+            self.set_counter(dvs, obj_id, pg_drop_attr, val)
+
+    def verify_value(self, dvs, obj_ids, entry_name, expected_value):
+        counters_db = swsscommon.DBConnector(swsscommon.COUNTERS_DB, dvs.redis_sock, 0)
+        table = swsscommon.Table(counters_db, "COUNTERS")
+
+        for obj_id in obj_ids:
+            ret = table.get(obj_id)
+            status = ret[0]
+            assert status
+            keyvalues = ret[1]
+            found = False
+            for key, value in keyvalues:
+              if key == entry_name:
+                  assert value == expected_value, "Saved value not the same as expected"
+                  found = True
+            assert found, "no such entry found"
+
+    def get_oids(self, dvs, obj_type):
+        db = swsscommon.DBConnector(swsscommon.ASIC_DB, dvs.redis_sock, 0)
+        tbl = swsscommon.Table(db, "ASIC_STATE:{0}".format(obj_type))
+        keys = tbl.getKeys()
+        return keys
+
+    def set_up_flex_counter(self, dvs):
+        for pg in self.pgs:
+            dvs.runcmd("redis-cli -n 5 hset 'FLEX_COUNTER_TABLE:PG_DROP_STAT_COUNTER:{}' ".format(pg) + "PG_COUNTER_ID_LIST '{}'".format(pg_drop_attr))
+
+        dvs.runcmd("redis-cli -n 4 hset 'FLEX_COUNTER_TABLE|PG_DROP' 'FLEX_COUNTER_STATUS' 'enable'")
+
+        self.populate_asic(dvs, "0")
+
+
+    def set_up(self, dvs):
+        self.pgs = self.get_oids(dvs, "SAI_OBJECT_TYPE_INGRESS_PRIORITY_GROUP")
+
+        db = swsscommon.DBConnector(swsscommon.COUNTERS_DB, dvs.redis_sock, 0)
+        tbl = swsscommon.Table(db, "COUNTERS_QUEUE_TYPE_MAP")
+
+        self.uc_q = []
+        self.mc_q = []
+
+        for q in self.qs:
+             if self.qs.index(q) % 16 < 8:
+                 tbl.set('', [(q, "SAI_QUEUE_TYPE_UNICAST")])
+                 self.uc_q.append(q)
+             else:
+                 tbl.set('', [(q, "SAI_QUEUE_TYPE_MULTICAST")])
+                 self.mc_q.append(q)
+
+    def test_pg_drop(self, dvs):
+        self.pgs = self.get_oids(dvs, "SAI_OBJECT_TYPE_INGRESS_PRIORITY_GROUP")
+        self.set_up_flex_counter(dvs)
+
+        self.populate_asic(dvs, "100")
+        time.sleep(self.DEFAULT_POLL_INTERVAL + 1)
+        self.verify_value(dvs, self.pgs, pg_drop_attr , "100")
+
+        self.populate_asic(dvs, "123")
+        time.sleep(self.DEFAULT_POLL_INTERVAL + 1)
+        self.verify_value(dvs, self.pgs, pg_drop_attr, "123")
```

Comment:
please put this in try-finally and clean up the flex counter

Reply:
Done